### PR TITLE
feat(core): implement wellknown swagger endpoints

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Implement well-known swagger endpoints.

This PR implements the `management.openapi.json` and `experience.openapi.json` swagger endpoints under the `well-known` path.  
The original `swagger.json` endpoint will be used as a fallback swagger endpoint that includes all the endpoints defined in Logto. 
In contrast, the newly created well-known swagger endpoints will be categorized based on the API usage scenario. E.g management API and experience API.  

- Restructure the current well-known router files, and group them under a `well-known` path.
- Add new `well-known.openapi.ts` router file. 
- Extract common util methods from the `swagger.json` endpoint to a newly created `swagger/utils/operaction.ts` file. So we may keep a single source of truth for those swagger build methods for both `swagger` endpoints and `well-known.openapi` endpoints. 
- Also fork the old `well-known/sign-in-exp` endpoint with a new `well-known/experience` endpoint. 

image

## Testing

test locally

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/init.ts [hunk 1]
```diff
@@ -45,7 +45,8 @@ import systemRoutes from './system.js';
 import type { AnonymousRouter, ManagementApiRouter } from './types.js';
 import userAssetsRoutes from './user-assets.js';
 import verificationCodeRoutes from './verification-code.js';
-import wellKnownRoutes from './well-known.js';
+import wellKnownRoutes from './well-known/index.js';
+import wellKnownSwaggerRoutes from './well-known/well-known.openapi.js';
```
Comment (gao-sun) [inline packages/core/src/routes/init.ts:49]: should align the naming to `wellKnownOpenApiRoutes`?

File: packages/core/src/routes/swagger/utils/general.ts [hunk 2]
```diff
@@ -293,3 +313,12 @@ export const pruneSwaggerDocument = (document: OpenAPIV3.Document) => {
 
   prune(document);
 };
+
+/**
+ * Check if the given router is a management API router. The function will check if the router
```
Comment (gao-sun) [inline packages/core/src/routes/swagger/utils/general.ts:318]:
```suggestion
 * Check if the given router is a Management API router. The function will check if the router
```

File: packages/core/src/routes/swagger/utils/general.ts [hunk 3]
```diff
@@ -293,3 +313,12 @@ export const pruneSwaggerDocument = (document: OpenAPIV3.Document) => {
...
+ * contains koa-auth middleware.
```
Comment (gao-sun) [inline packages/core/src/routes/swagger/utils/general.ts:319]:
```suggestion
 * contains the `koaAuth` middleware.
```

File: packages/core/src/routes/well-known/well-known.openapi.ts [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    });
+
+    const allSupplementDocuments = await Promise.all(
+      supplementPaths.map(async (path) =>
+        removeUnnecessaryOperations(
+          // eslint-disable-next-line no-restricted-syntax -- trust the type here as we'll validate it later
+          JSON.parse(await fs.readFile(path, 'utf8')) as DeepPartial<OpenAPIV3.Document>
+        )
+      )
+    );
```
Comment (gao-sun) [inline packages/core/src/routes/well-known/well-known.openapi.ts:220]: copy - paste here? can we do a util?

Comment (simeng-li) [reply packages/core/src/routes/well-known/well-known.openapi.ts]: Updated extract a `getSupplementDocuments` util method

**Review by gao-sun — APPROVED** [review]

**Pull request merged**